packages/backend/src/apps/telegram-bot/common/interceptor/rate-limit.ts

```diff
@@ -0,0 +1,32 @@
+import type { IApp } from '@plumber/types'
+
+import logger from '@/helpers/logger'
+
+const rateLimitHandler: IApp['requestErrorHandler'] = async function (
+  $,
+  error,
+) {
+  if (error.response.status !== 429) {
+    return // use default error handling
+  }
+
+  const retryAfter = +error.response.headers['retry-after']
```

Comment:
nit: not needed in this PR, but i'm wondering if we should make it a team coding style to use Axio's [`get`](https://github.com/axios/axios?tab=readme-ov-file#axiosheadersgetheader) instead of directly accessing headers. Reason is that axios [normalizes](https://github.com/axios/axios/blob/6d4c421ee157d93b47f3f9082a7044b1da221461/lib/core/AxiosHeaders.js#L111) header names for us when using `get`. Could prevent some unintended bugs later on.

```suggestion
  const retryAfter = +error.response.headers['retry-after']
```

Reply:
Actually nvm, it doesn't typecheck =.=. sigh. Maybe we should lint to only use lower case header names, since it looks like axios [does this](https://github.com/axios/axios/blob/6d4c421ee157d93b47f3f9082a7044b1da221461/lib/core/AxiosHeaders.js#L83) each time it constructs `AxiosHeaders`.